File: frontend/src/app/shared/models/auth.models.ts
export interface UserProfile {
  id: string;
  email: string;
  displayName: string;
  roles: string[];
}

export interface LoginResponse {
  token: string;
  expiresAt: string;
  user: UserProfile;
}

export interface StaffUser {
  id: string;
  email: string;
  displayName: string;
  roles: string[];
  isActive: boolean;
}

export const AppRole = {
  Waiter: 'Waiter',
  Manager: 'Manager',
  Kitchen: 'Kitchen',
  Admin: 'Admin',
} as const;

export type AppRoleName = (typeof AppRole)[keyof typeof AppRole];

File: src/Resto.Application/Auth/Models/AuthModels.cs
namespace Resto.Application.Auth.Models;

public sealed record LoginResult(
    string Token,
    DateTime ExpiresAt,
    UserProfile User);

public sealed record UserProfile(
    Guid Id,
    string Email,
    string DisplayName,
    IReadOnlyList<string> Roles);

public sealed record StaffUserDto(
    Guid Id,
    string Email,
    string DisplayName,
    IReadOnlyList<string> Roles,
    bool IsActive);

public sealed record CreateStaffUserRequest(
    string Email,
    string Password,
    string DisplayName,
    string Role);
